src/main/resources/assets/js/application/recruiter/services/tdprScheduleService.js

```diff
@@ -0,0 +1,51 @@
+define(['application/recruiter/tdprRecruiterModule', 'application/recruiter/services/tdprRecruiterSlotsService', 'application/recruiter/services/tdprDateService'], function (tdprRecruiterModule) {
+    tdprRecruiterModule.service('tdprScheduleService', ['tdprRecruiterSlotsService', 'AvailabilityEnum', 'dateFilter', function (tdprRecruiterSlotsService, AvailabilityEnum, dateFilter) {
+        var service = {};
+
+        service.changeSlotType = function (slotId, day, person, changeTo) {
```

Comment:
This function should be moved to controller and called from slot directive so we won't need filter functions